frontend/src/store/authStore.ts

```diff
@@ -43,36 +44,45 @@ export const useAuthStore = defineStore('auth', () => {
 
     let kc: Keycloak = new Keycloak(initOptions);
 
+    // Once KC is set up and connected flag it as 'ready'
     kc.onReady = function (authenticated) {
       ready.value = true;
     };
 
+    // After a refresh token fetch success
+    kc.onAuthRefreshSuccess = function () {
+      // console.log($keycloak.value.token);
+    };
+
     await kc
       .init({ onLoad: 'check-sso', pkceMethod: 'S256' })
       .then(() => {
         // Set the state field to the inited keycloak instance
         $keycloak.value = kc;
 
-        //Token Refresh
+        // Token Refresh
+        // Check token validity every 10s and, if necessary, update the token.
+        // Refresh token if it's valid for less then 70 seconds
         setInterval(() => {
-          $keycloak.value
-            .updateToken(70)
+          kc.updateToken(70) // If the token expires within 70 seconds from now get a refreshed
```

Comment:
1) I can't remember, but we should make sure this setInterval is saved into auth state somehow. IIRC, setInterval returns an integer which is an identifier for the periodic function. We'll want to make sure that even though we don't use `clearInterval(identityId)`, we can choose to do so in the future if we need to by having it be assigned somewhere pragmatic in the auth store. Perhaps we should have it be called like `const refreshJob = setInterval((...` or something?

2) Magic numbers like that 70 seconds should be constant defined in a constants.js file.

Reply:
I saved the number in state (needed to change to window.setInterval or else it returns some other type that TS is wierd about https://stackoverflow.com/questions/53189729/typescript-error-setinterval-type-timer-is-not-assignable-to-type-number)
Let me know it that's what you had in mind?